addons/account/models/product.py

```diff
@@ -118,6 +118,22 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model
+    def _force_default_tax(self, products, companies):
+        if not products:
+            # in case we're calling that function from demo data file, we want to force the default taxes on all the templates
+            products = self.env['product.template'].search([])
+
+        default_customer_tax_ids = companies.filtered('account_sale_tax_id').account_sale_tax_id.ids
+        default_supplier_tax_ids = companies.filtered('account_purchase_tax_id').account_purchase_tax_id.ids
+
+        products_without_company = products.filtered(lambda p: not p.company_id).sudo()
+        products_without_company.taxes_id = [Command.link(tax) for tax in default_customer_tax_ids]
+        products_without_company.supplier_taxes_id = [Command.link(tax) for tax in default_supplier_tax_ids]
```

Comment:
Same comment as before 🤷
```suggestion
        default_customer_taxes = companies.filtered('account_sale_tax_id').account_sale_tax_id
        default_supplier_taxes = companies.filtered('account_purchase_tax_id').account_purchase_tax_id

        products_without_company = products.filtered(lambda p: not p.company_id).sudo()
        products_without_company.taxes_id += default_customer_taxes
        products_without_company.supplier_taxes_id += default_supplier_taxes
```

Reply:
what can I say? I like copy/pasta :shrug: